### PR TITLE
Add instance loss

### PR DESCRIPTION
**Pull request by layumi**
Hi Kevin, 
Thanks for the great repo. I add the code for instance loss. 
I am not sure whether the code style is good. 
If any modification is needed, please let me know. Thanks a lot.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks! I will try to take a look and merge this for version 1.2

**Comment by layumi** [comment]
Hi @KevinMusgrave 
Sorry. I just forgot to provide the paper link. 
https://dl.acm.org/doi/10.1145/3383184

**Comment by KevinMusgrave** [comment]
Thanks @layumi

**Comment by KevinMusgrave** [comment]
@layumi is there a reference implementation that we can add as a unit test?

**Comment by layumi** [comment]
Hi @KevinMusgrave 


Thanks a lot. 

1) Is there a unit test example? I can follow it to add one for instance loss.  

2) In my repo, the instance loss can be used like other losses in your repo. The behaviour is similar to losses.contrastive loss,  losses.triplet loss.   Hope it helps to explain the implementation.

You may refer to 
https://github.com/layumi/Person_reID_baseline_pytorch/blob/master/train.py#L218-L221

**Comment by KevinMusgrave** [comment]
You can add a new test file: `tests/losses/test_instance_loss.py`. The bare minimum test should be to make sure the loss funtion runs without crashing. Something like this:

```python
import unittest
import torch

from pytorch_metric_learning.losses import InstanceLoss

from .. import TEST_DEVICE, TEST_DTYPES

class TestInstanceLoss(unittest.TestCase):
    def test_instance_loss(self):
        loss_fn = InstanceLoss()
        for dtype in TEST_DTYPES:
            embeddings = torch.randn(32, 128, device=TEST_DEVICE, dtype=dtype)
            labels = torch.randint(0, 10, size=(32,), device=TEST_DEVICE)
            loss = loss_fn(embeddings, labels)
```

Then run `python -m unittest tests/losses/test_instance_loss.py`. 

If you find that it's too difficult to extend BaseMetricLossFunction, then just extend torch.nn.Module and I can take a look afterwards to see if it makes sense to extend BaseMetricLossFunction.

**Comment by layumi** [comment]
Hi @KevinMusgrave 

I just added the test code.

It works now. 
```bash
python -m unittest tests/losses/test_instance_loss.py 
```

**Comment by KevinMusgrave** [comment]
Thanks @layumi !
